### PR TITLE
Fix the size and position of overflow:scroll clip

### LINKED ISSUE #16576: Box with overflow:hidden cut inner div with absolute position by content edge, not by padding edge
Tested on Windows 8.1 x64

servo-overflow

```
<html>
	<body>
		<style>
			#box{
				background-color: #080b0c;
				height: 100px;
				overflow: hidden;
				padding: 0 15px;
				position: relative;
			}
			#inside {
				position: absolute;
				right: 0px;
				width: 100px;
				height: 100%;
				background: red;
			}
		</style>
		<div id="box">
			<div id="inside">
			</div>
		</div>
	</body>
</html>
```

### PR DESCRIPTION
**Pull request by mrobinson**
The clip for overflow:scroll clip was not including the space for
padding, which could lead to some content being clipped when it
shouldn't. This changes fixes that issue.

A test is skipped because this fix also stops hiding a failure due to
mispositioned text.

Fixes #16576.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #16576 (github issue number if applicable).

- [x] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17712)

### CODE REVIEW

File: tests/wpt/metadata-css/css-backgrounds-3_dev/html4/attachment-local-clipping-image-6.htm.ini [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[attachment-local-clipping-image-6.htm]
```
Comment (emilio) [inline tests/wpt/metadata-css/css-backgrounds-3_dev/html4/attachment-local-clipping-image-6.htm.ini]: Could you file an issue with details on why this has regressed, and maybe point to it from the ini file?

**Review by emilio — COMMENTED** [review]
Looks reasonable, though I'm curious why that test has regressed

File: tests/wpt/metadata-css/css-backgrounds-3_dev/html4/attachment-local-clipping-image-6.htm.ini (hunk 1 above)
Comment (mrobinson) [inline tests/wpt/metadata-css/css-backgrounds-3_dev/html4/attachment-local-clipping-image-6.htm.ini]: It's not really a regression, just a failure that was hidden by the bug this PR fixes. I think I know what the issue is though and hope to fix it in a later PR.

Comment (mrobinson) [inline tests/wpt/metadata-css/css-backgrounds-3_dev/html4/attachment-local-clipping-image-6.htm.ini]: Here is the issue which I think is causing the failure in this test: https://github.com/servo/servo/issues/17718

Comment (emilio) [inline tests/wpt/metadata-css/css-backgrounds-3_dev/html4/attachment-local-clipping-image-6.htm.ini]: Thanks!

**Pull request merged**